doc/processes/release_process.md

```diff
@@ -0,0 +1,109 @@
+[2] http://openjdk.java.net/legal/assembly-exception.html
+
+SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# Eclipse OpenJ9 Release Process
+
+The OpenJ9 JVM is composed of two code repositories:
+
+* OpenJ9 [https://github.com/eclipse/openj9]()
+* OMR [https://github.com/eclipse/openj9-omr]()
+
+OpenJ9 needs to be combined with code from OpenJDK to create a Java SDK.
+The OpenJDK code is managed in separate github repos referred to as the 
+"OpenJDK extensions for OpenJ9 for JDK X".  Currently there are two 
+extensions repos:
+
+* JDK8 [https://github.com/ibmruntimes/openj9-openjdk-jdk8]()
+* JDK9 [https://github.com/ibmruntimes/openj9-openjdk-jdk9]()
+
+OpenJ9 binaries are built by the AdoptOpenJDK community.  
+
+An OpenJ9 release should:
+
+* Use a single code base to support the various JDK-levels.
+* Track the upstream OpenJDK releases.  
+* Use a consistent set of tags across OpenJ9 & extensions repos to 
+ensure that releases can be rebuilt or branched later.
+* OpenJ9 releases should be high quality and pass a common quality bar. 
+* Communicate the tags for a release to AdoptOpenJDK to ensure that
+binaries at the correct levels have been created. 
+* Clearly define the supported platforms
+* Should not regress performance from release to release.
+
+
+## Release cadence
+OpenJ9 should aim to release quarterly to match the OpenJDK long term 
+support (LTS) release cadence.  OpenJDK 8 is released quarterly and 
+other LTS releases will likely follow similar release cadences.  There 
+should be monthly milestone builds as the project builds towards the 
+next release.
+
+## Quality bar
+Work with the AdoptOpenJDK community to define a common quality bar for 
+any OpenJDK-based JDK.  The aim is to define a set of tests, both 
+functional and stress, and 3rd party applications that should pass for 
+every release.
+
+## Platform support
+Define the platforms that are supported by this release.  The set of 
+platforms will depend on the availability of hardware at OpenJ9 and 
+AdoptOpenJDK that can be used to validate the platform meets the quality 
+bar.  The code base may support more platforms than can be validated by 
+the quality.  Such platforms will not be officially included in the release.
+
+## Performance
+OpenJ9 should avoid regressing performance from release to release.
+This is harder to validate in the open project given the limitted 
+hardware resources available.  Work is being done at AdoptOpenJDK to 
+enable performance testing.
+
+# Release Process
+
+1. Create an issue to track the release and record the information required
+above, including the SHAs and tags.  This information will mostly be duplicated
+with the Eclipse release plan.
+1. Ask committers - via Slack and mailing list -  to refrain from committing any
+further changes until an "all clear" signal is broadcast.  Disable the promotion
```

Comment:
How are we going to make sure the freezes are of short duration? The danger with freezing the codebase is that PR queues grow and the community's development efforts stall while we release. Since we can branch (as the process described below details) why do we need a freeze when history can't change?

Reply:
I also see little value in freezing the code just to create some branches (or tags).

I despise code freezes.

Reply:
I too despise code freezes.  My thinking here was there are a number of projects and repos involved and having a short window where the world isn't changing under our feet may make it easier to get through the process, especially if something goes wrong.

Today, it's 4 repos that need to be coordinated and it may not be immediately clear which SHAs in each repo should be used for the release.  As far as I am aware, there is no guarantee that the Adopt nightly builds have run on the exact same commit for all elements in the matrix of JDK releases x platforms.

Having a window where nothing goes in makes it possible to ensure that Adopt is building the same code on all releases x platforms to give the initial set of SHAs